spec/system/consumer/authentication_spec.rb

```diff
@@ -156,6 +156,30 @@
       end
     end
 
+    describe "Logging in from the private shop page" do
+      let(:distributor) { create(:distributor_enterprise, require_login: true) }
+      let!(:order_cycle) {
+        create(:simple_order_cycle, distributors: [distributor],
+                                    coordinator: create(:distributor_enterprise))
+      }
+      before do
+        visit enterprise_shop_path(distributor)
+      end
+
+      it "clicking Login triggers the login modal" do
+        find("a", text: "login").click
+        expect(page).to have_selector("a.active", text: "Login")
+        expect(page).to have_button("Login")
+      end
+
+      # needs to be changed once #8860 is addressed
+      xit "clicking Signup triggers login modal and opens the signup tab" do
+        find("a", text: "signup").click
+        expect(page).to have_selector("a.active", text: "Sign up")
+        expect(page).to have_button "Sign up"
```

Comment:
This one will fail until #8860 is ready.
We should test for `have_button "Login"` ...